### PR TITLE
Backend Skeleton

### PR DESCRIPTION
**Pull request by esheehan-gsl**
Create a skeleton for the Flask API we'll use to provide data to the user interface.

## Overview

This PR adds a Python web application that will serve as the backend for serving diagnostics data. At the moment, it only responds with a test JSON message:

```json
{ "msg": "Hello, Dave" }
```

There is a Dockerfile which creates an image that runs the Flask application with Gunicorn.

```sh
cd services/api
docker build -t unified-graphics/api:latest .
docker run --rm -p 5000:80 unified-graphics/api
```

Then you should be able to load http://localhost:5000 and see the above JSON response in your browser.

## To Do
- [x] Configure tests with pytest
- [x] Configure type checking with mypy
- [x] Configure Black for formatting
- [x] Provide a simple JSON response for testing
- [x] Define a Docker image for the cluster
- [x] Configure a pre-commit hook to lint everything
- [x] Add `CONTRIBUTING.md` to explain how to run everything

### CODE REVIEW

**Comment by esheehan-gsl** [comment]
@ian-noaa I’m not quite done here, because I need to update the tests and I’d like to add automatic linting with a git pre-commit hook, but there is now a Dockerfile available, if you’d like to take a look and experiment with running cluster locally.

If you want to run any of the tools (tests, linters, <i>etc.</i>), you’ll need to [install Poetry](https://python-poetry.org/docs/#osx--linux--bashonwindows-install-instructions). Then you can run

```sh
cd services/api
poetry run flake8 .
poetry run mypy .
poetry run pytest
```

**Comment by esheehan-gsl** [comment]
Oh, and you can ignore the changes to files in `services/ui`. When I installed Black for formatting the Python, I updated our Prettier config so that it formats our JavaScript with the same conventions and ran it over all of the files there.

**Comment by ian-noaa** [comment]
Sounds great! I'm looking forward to using Poetry on this project. Are you managing your local Python version with pyenv?

And thanks for the Dockerfile - it'll be nice to have the frontend and the backend to experiment with in the local cluster.

**Comment by esheehan-gsl** [comment]
Yeah, I think I am using `pyenv`. Unless poetry has support for installing different versions of Python, it’s probably integrating with pyenv.

File: .githooks/pre-commit [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+# Globals:
+#   PROJ_ROOT
+# Arguments:
+#   File to check, a path
+#   Type of file being checked (js, cjs, svelte, css, or scss)
+########################
+check_web() {
+  # FIXME: We should probably determine the service directory dynamically from
+  # the file path instead of hard-coding. We could conceivably end up with
+  # multiple Python services.
```
Comment (ian-noaa) [inline .githooks/pre-commit:54]: A small nit - the `check_web()` function works on node/js services, and not Python ones.

```suggestion
  # multiple Node services.
```

**Review by ian-noaa — APPROVED** [review]
This looks great. The `Contributing.md` is particularly helpful and the git hooks will be awesome so we can avoid adding "formatting fixes" commits.

I'll go ahead and update the Kubernetes manifests to connect to port 80 on the pods. I like the idea of standardizing our services on one port - it'll be easier than having each service on a bespoke port. Since we're using containers, we have control over how those ports are exposed anyways so we won't have port conflicts.

**Pull request merged**